### PR TITLE
Moves languages to a holder datum, gives mind its own language holder

### LINKED ISSUE #26682: Librarians who die and get cloned lose their languages
I am aware this bug exists, I'm working on fixing it.

### PR DESCRIPTION
**Pull request by coiax**
Fixes #26682.
Contains changes from #26650.

:cl: coiax
fix: If you know more than the usual number of languages, you'll now remember them after you get cloned.
del: Humans turning into monkeys will not suddenly be able to speak the monkey language, and will continue to understand and speak human.
add: Ghosts can now modify their own understood languages with the
language menu.
add: Any mob can also open their language menu with the IC->Open Language Menu verb.
add: Silicons can now understand Draconic as part of their internal databases.
/:cl:

- All atoms now have a single var for storing language details, `language_holder`.
- All language procs get this holder when manipulating languages.
- Living mobs, if they possess a mind, return use mind's `language_holder` instead for all language operations.
- When a mind is transfered into a mob, if it does not have any existing `language_holder`, it copies the information from the mob's existing holder. This will mean that fresh minds in swarmer bodies will inherit the swarmer's language details.
- New proc `could_speak_in_language` which determines if the mob COULD speak in a language, ignoring checks like OMNITONGUE or them not knowing the language in the first place. This makes things a lot simpler.
- Language holders have a `shadow_languages` var, which are a list of languages that the mob grants by occupying, but do not transfer with the mind.

### Bugs

- [x] Silicons no longer have machine language.
- [x] Ghost roles, and spawned syndicate cyborgs are unable to understand any languages
- [x] The language menu could not be opened for a syndicate cyborg
- [x] AIs sometimes randomly forget all languages

- [x] Sometimes the language holder disappears, and I don't know why

### CODE REVIEW

**Comment by octareenroon91** [comment]
Will this break snpcs?

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -47,8 +46,7 @@
 
 /atom/movable/Initialize(mapload)
 	. = ..()
-	for(var/L in initial_languages)
-		grant_language(L)
+	language_holder = new initial_language_holder(src)
```
Comment (octareenroon91) [inline code/game/atoms_movable.dm]: Is this going to drive up memory bloat to a noticeable degree?

Comment (coiax) [inline code/game/atoms_movable.dm]: hopefully not, but I don't know.

**Comment by Anonmare** [comment]
This breaks silicon languages I think. They only speak common now.

**Comment by Anonmare** [comment]
Also the AI forgot common

**Comment by Thunder12345** [comment]
# *****TEST YOUR FUCKING SHIT*****

**Comment by octareenroon91** [comment]
@Anonmare consider these two parts together:

````
 /mob/living/silicon
//[ . . . ]
	initial_language_holder = /datum/language_holder/synthetic 
````

````
/datum/language_holder/synthetic
	initial_languages = list(/datum/language/common, /datum/language/machine)
````

File: code/game/atoms_movable.dm (hunk 1 above)
Comment (octareenroon91) [inline code/game/atoms_movable.dm]: That's probably worth testing.

**Comment by coiax** [comment]
@octareenroon91 the bug is likely to do with how the initial round spawn makes everyone a human, then transforms them into different mobs

**Comment by octareenroon91** [comment]
Oh, there was a testmerge and shitstorm?

**Comment by Niknakflak** [comment]
Why did you close your other PR where everyone hated that ghost change and called it dumb, just to open this PR with a bunch of other shit piled on.  

This is hilariously manipulative and I encourage pressure to atomize the PR or at least in my opinion, JUST HAVE GHOSTS UNDERSTAND EVERYTHING ALL THE TIME

Who wants to fuck with a menu to get the proper conversation while being dead.

**Comment by Niknakflak** [comment]
My statement still stands

**Comment by coiax** [comment]
This has been tested for a number of rounds, and no bugs are immediately apparent.

File: code/modules/language/language_holder.dm [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	copy.selected_default_language = src.selected_default_language
+	// language menu is not copied, that's tied to the holder.
+	copy.omnitongue = src.omnitongue
+	return copy
+
+/datum/language_holder/proc/grant_language(datum/language/dt)
+	languages[dt] = TRUE
+
+/datum/language_holder/proc/grant_all_languages(omnitongue=FALSE)
+	for(var/la in subtypesof(/datum/language))
```
Comment (AnturK) [inline code/modules/language/language_holder.dm]: There's few places you use subtypesof(/datum/language), please cache it, having an all languages list will be useful.

**Comment by GunHog** [comment]
Love it, love it, except for one thing: Monkey-humans should speak and understand only monkey. I believe this is a deliberate thing to require a translator to communicate. Most player monkeys occur from genetics, do they not? The monkey language would be greatly reduced in usage if genetics-made monkeys were not forced to use it. 

The monkey language is a _limitation_, not an advantage. It is so that, say, a geneticist tricks you into taking a monkey syringe, you cannot simply run away and yell at the nearest doctor to explain what happened. I recommend that it be preserved as it is now for this particular language.

**Comment by coiax** [comment]
@GunHog it would be very difficult to enforce forgetting a language without a host of special casing.

What about, after this PR is merged, give monkeys their own tongues which can only speak monkey.

**Comment by GunHog** [comment]
Seems perfect to me!

**Comment by coiax** [comment]
@AnturK I noticed some runtimes when testing, probably due to the lazy initialization of language holders. I've fixed them, but I'd hold off on merging for now, review it if you want, but I'd prefer to get another round of test merges in, just to triple check there's nothing I've missed.

**Comment by Niknakflak** [comment]
Ghost language panel that everyone thinks is dumb is still in

**Comment by AnturK** [comment]
Alright, no problem.

**Pull request merged**